### ISSUE TITLE
`assert(something ~= false)` easier to read as `assert(something)`?

### ISSUE THREAD

**Issue by hughperkins** [Element-Research/rnn#102]
In a couple of places, there are double-negatives like:

https://github.com/Element-Research/rnn/blob/master/Recursor.lua#L22
```
 if self.train ~= false then
```
and https://github.com/Element-Research/rnn/blob/master/Recursor.lua#L130
```
 assert(online ~= false, ...
```
I find these really hard to read, and have to double-take on them for a while :-)  I think they can be written simply as the affirmative? :
```
 if self.train then
```
```
 assert(online, ...
```

**Comment by nicholas-leonard** [comment]
@hughperkins Sorry for confusion. The `if self.train ~= false then` is used to handle `self.train = nil` (the default value of `self.train`. So it is equivalent to `if self.train or self.train == nil then`. Same principle for the the `assert(online ~= false)`. The default value of online (i.e. when `online = nil`) is true. 

I have just added comments to the code to make this more obvious. Hope that helps.

**Comment by hughperkins** [comment]
Ah... can we maybe add to the `__init` function, or similar, something like:

```
self.train = self.train or true
```
?

**Comment by hughperkins** [comment]
hmmm, oh wait, that doesnt work, right?  hmmm...

**Comment by hughperkins** [comment]
Seems like `~= false` is a good way of doing this.  I guess we only need to do it once perhaps, in `__init` method?

```
self.train = self.train ~= false
```

**Comment by nicholas-leonard** [comment]
@hughperkins Ideally, this would be added to the [`nn.Module:__init`](https://github.com/torch/nn/blob/master/Module.lua#L3). What do you think @soumtih ?

**Comment by hughperkins** [comment]
Well... for `Dropout`, which is directly comparable, since behavior changes between train and evaluate, they are putting it in `Dropout.__init`:

https://github.com/torch/nn/blob/master/Dropout.lua#L6
```
 self.train = true
```
Hmmm... although... now I check the `git blame`, I see that... you wrote that :-)

```
b7822796 (nicholas-leonard 2014-07-09 14:26:32 -0400  6)    self.train = true
```

**Comment by hughperkins** [comment]
Created a PR, see if it gets merged https://github.com/torch/nn/pull/553
